### PR TITLE
test(wrapper): add unit test of schema wrapper

### PR DESCRIPTION
**Pull request by Zippo-Wang**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestSchemaWrapperGet
--- PASS: TestSchemaWrapperGet (0.00s)
=== RUN   TestPrimToArray
--- PASS: TestPrimToArray (0.00s)
=== RUN   TestListToSlice
--- PASS: TestListToSlice (0.00s)
=== RUN   TestListToObject
--- PASS: TestListToObject (0.00s)
=== RUN   TestListToArray
--- PASS: TestListToArray (0.00s)
=== RUN   TestConvArrayType
--- PASS: TestConvArrayType (0.00s)
=== RUN   TestSetToArray
--- PASS: TestSetToArray (0.00s)
=== RUN   TestSetToSlice
--- PASS: TestSetToSlice (0.00s)
=== RUN   TestSetToObject
--- PASS: TestSetToObject (0.00s)
=== RUN   TestMapToStrMap
--- PASS: TestMapToStrMap (0.00s)
=== RUN   TestMapToMap
--- PASS: TestMapToMap (0.00s)
PASS

coverage: 8.1% of statements in ../../../../terraform-provider-huaweicloud/...
```

### CODE REVIEW

File: huaweicloud/helper/schemas/schema_wrapper_test.go [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+	assert.Equal(t, nil, w.Get("money"))
+	assert.Equal(t, 0.0, w.Get("money", true))
+	assert.Equal(t, nil, w.Get("money", false))
+
+	assert.Equal(t, nil, w.Get("gender"))
+	assert.Equal(t, false, w.Get("gender", true))
+	assert.Equal(t, nil, w.Get("gender", false))
+
+	assert.Equal(t, nil, w.Get("suke"))
+	assert.Equal(t, nil, w.Get("suke", false))
```
Comment (chengxiangdong) [inline huaweicloud/helper/schemas/schema_wrapper_test.go:134]:
```suggestion
	assert.Equal(t, nil, w.Get("suke"))
	assert.Equal(t, nil, w.Get("suke", false))
	assert.Equal(t, fmt.Sprintf("%#v", &schema.Set{}), fmt.Sprintf("%#v", w.Get("suke", true)))
```

File: huaweicloud/helper/schemas/schema_wrapper_test.go [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+	assert.Equal(t, nil, w.PrimToArray("money", false))
+
+	assert.Equal(t, nil, w.PrimToArray("gender"))
+	assert.Equal(t, []bool{}, w.PrimToArray("gender", true))
+	assert.Equal(t, nil, w.PrimToArray("gender", false))
+
+	assert.Equal(t, nil, w.PrimToArray("foo"))
+	assert.Equal(t, []any{}, w.PrimToArray("foo", true))
+	assert.Equal(t, nil, w.PrimToArray("foo", false))
+
```
Comment (chengxiangdong) [inline huaweicloud/helper/schemas/schema_wrapper_test.go:197]:
```suggestion

	assert.Equal(t, nil, w.PrimToArray("vpcs"))
	assert.Equal(t, []any{}, w.PrimToArray("vpcs", true))

```

File: huaweicloud/helper/schemas/schema_wrapper_test.go (hunk 1 above)
Comment (Zippo-Wang) [reply huaweicloud/helper/schemas/schema_wrapper_test.go:134]: fix

File: huaweicloud/helper/schemas/schema_wrapper_test.go (hunk 2 above)
Comment (Zippo-Wang) [reply huaweicloud/helper/schemas/schema_wrapper_test.go:197]: fix

**Comment by chengxiangdong** [comment]
LGTM
/approve

File: huaweicloud/helper/schemas/schema_wrapper_test.go [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+	}))
+	assert.Equal(t, []interface{}{}, w.ListToSlice("foo", true, func(i int) any {
+		return d.Get(fmt.Sprintf("foo.%d", i))
+	}))
+
+	// init data
+	InitTestStruct(t, d)
+	assert.Equal(t, nil, w.ListToSlice("name", false, func(_ int) any { return d.Get("name") }))
+	assert.Equal(t, nil, w.ListToSlice("beta", false, func(_ int) any { return d.Get("beta") }))
+	assert.Equal(t, nil, w.ListToSlice("beta", false, func(_ int) any { return d.Get("beta") }))
```
Comment (deer-hang) [inline huaweicloud/helper/schemas/schema_wrapper_test.go]: This line of code is duplicated with line 252.

Comment (Zippo-Wang) [reply huaweicloud/helper/schemas/schema_wrapper_test.go]: fix

**Pull request merged**